main.tf

```diff
@@ -72,6 +72,11 @@ resource "aws_launch_configuration" "main" {
   user_data            = var.user_data
   user_data_base64     = var.user_data_base64
 
+  metadata_options {
+    # Require use of IMDSv2
+    http_tokens = var.imdsv2 == true ? "required" : "true"
```

Comment:
Seems like the default value here should be `optional` per https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#metadata-options

Nit: Since this is a boolean, you don't need to explicitly check for `true`. `var.imdsv2 ? "required" : "optional"` will also work

Reply:
That's a fatfinger on my part 🤦‍♂️ Thanks!